app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -0,0 +1,110 @@
+/** @file Provides {@link Cognito} class which is the entrypoint into the AWS Amplify library.
+ *
+ * All of the functions used for authentication are provided by the AWS Amplify library, but we
+ * provide a thin wrapper around them to make them easier to use. Mainly, we perform some error
+ * handling and conditional logic to vary behaviour between desktop & cloud. */
+import * as amplify from '@aws-amplify/auth'
+import * as cognito from 'amazon-cognito-identity-js'
+import * as results from 'ts-results'
+
+import * as config from './config'
+
+// ===============
+// === Cognito ===
+// ===============
+
+export interface Cognito {
```

Comment:
is there a need for a separate interface?
- typescript is structurally typed so you can do `implements CognitoImpl`
- alternatively if it's because there will be more implementations added later you might want to be more descriptive with the naming of `CognitoImpl`